Sources/Sham/Models/StubRequest.swift

```diff
@@ -110,17 +110,46 @@ public struct StubRequest: CustomStringConvertible {
         
         return validScheme && validHost && validPort && validPath && validQuery
     }
-}
-
-// MARK: - Hashable and Equatable
-
-extension StubRequest: Hashable, Equatable {
-    public func hash(into hasher: inout Hasher) {
-        hasher.combine(self.description)
-    }
     
-    public static func == (lhs: Self, rhs: Self) -> Bool {
-        return lhs.hashValue == rhs.hashValue
+    /// A function that generates a score to represent how well a stored stubbed request matches an incoming request
+    /// - Parameter request: The incoming request to generate the priority score against
+    /// - Returns: A score representing how well this request matches an incoming request.  Nil values imply no match.
+    public func priorityScore(for request: StubRequest) -> Int? {
+        guard self.canMockData(for: request) else {
+            return nil
+        }
+        
+        var score = 0
+        
+        if self.method == request.method {
+            score += 1
+        }
+        
+        guard let url = self.url else {
+            return score
+        }
```

Comment:
If I'm understanding this correctly, if the url matches, we return the score here... but wouldn't that return a low score for an exact match?

Reply:
this is saying if the request we are matching against the incoming request has no URL, return the current score.  is that how you were interpreting it? exact matches should have a URL.

Reply:
Oh no, my bad, sorry. I misread. Sounds good, you can resolve this!